dandiapi/api/migrations/0005_default_oauth_application.py

```diff
@@ -0,0 +1,42 @@
+# Generated by Django 3.0.9 on 2020-12-17 23:59
+# flake8: noqa N806
+
+from django.db import migrations
+from django.db.models import Q
+
+
+def create_application(apps, schema_editor):
+    Application = apps.get_model('oauth2_provider', 'Application')
+    name = 'DANDI GUI'
+    # This is specified so it matches the default value used in the frontend.
+    client_id = 'Dk0zosgt1GAAKfN8LT4STJmLJXwMDPbYWYzfNtAl'
+    if not Application.objects.filter(Q(name=name) | Q(client_id=client_id)).exists():
+        application = Application(
+            # Production instances should change this.
+            client_id='Dk0zosgt1GAAKfN8LT4STJmLJXwMDPbYWYzfNtAl',
+            # Production instances must change this.
+            redirect_uris='http://localhost:8085/',
+            # These values should not be modified.
+            client_type='public',
+            authorization_grant_type='authorization-code',
+            client_secret='',
+            name=name,
+            # This can be turned off in production if appropriate.
+            skip_authorization=False,
+        )
+        application.save()
+
+
+def reverse_create_application(apps, schema_editor):
+    # Deleting the application would destroy any customization applied to it.
```

Comment:
Just FYI, destroying customization is an expected result of reversing a migration, and shouldn't be a reason not to write a reverse migration.

However, reverse migrations aren't used very often, and it may not be worth the burden to write one.